src/main/kotlin/app/CodeLongevity.kt

```diff
@@ -0,0 +1,231 @@
+// Copyright 2017 Sourcerer Inc. All Rights Reserved.
+
+package app
+
+import org.eclipse.jgit.diff.DiffFormatter
+import org.eclipse.jgit.diff.DiffEntry
+import org.eclipse.jgit.diff.RawText
+import org.eclipse.jgit.internal.storage.file.FileRepository
+import org.eclipse.jgit.lib.ObjectId
+import org.eclipse.jgit.lib.ObjectLoader
+import org.eclipse.jgit.lib.Repository
+import org.eclipse.jgit.revwalk.RevCommit
+import org.eclipse.jgit.revwalk.RevWalk
+import org.eclipse.jgit.treewalk.TreeWalk
+import org.eclipse.jgit.util.io.DisabledOutputStream
+
+import java.util.Date
+
+/**
+ * Identifies a code line in a file in a revision.
+ */
+class RevCommitLine(val commit: RevCommit, val file: String, val line: Int)
+
+/**
+ * Represents a code line.
+ *
+ * TODO(Alex): the text arg is solely for testing proposes (remove it)
+ */
+class CodeLine(val from: RevCommitLine, val to: RevCommitLine, val text: String) {
```

Comment:
Since our goal is to compute a distribution of line ages per author, I don't believe you need this class at all. Your algo only adds entries to it, and your test code iterates through all CodeLine instances, without regard to from/to, etc. You could simply keep track of author->array of line ages.